### DISCUSSION TITLE
OctoDash Will Only Launch from Pi Desktop

### DISCUSSION THREAD

**Discussion by Keele8** [UnchartedBull/OctoDash#2181] [Q&A]
I recently installed OctoDash on a Raspberry Pi 4 with 4gb RAM and followed the steps outlined in Github. I also had issues getting it to launch which others had that were listed in the troubleshooting guide, but the solutions did not fix the issues. The issues were:

- From Troubleshooting page "Cannot open display, Xinit failing" (none of the fixes listed on the troubleshooting guide worked
- Octodash in the command terminal results in message "Selected language: EN" "Segmentation fault"
- export DISPLAY=0 doesn't fix issues
- Ratpoison in the command terminal results in message "can't open display"
- Reconfigured X11 server with sudo dpkg-reconfigure x11-common, did not fix anything
- xinit in he command terminal results in message "/usr/lib/xorg/Xorg.wrap: Only console users are allowed to run the X server
sudo xinit
xinit: giving up
xinit: unable to connect to X server: Connection refused
xinit: server error"
- sudo xinit in command line results in message "X.Org X Server 1.20.4
X Protocol Version 11, Revision 0
Build Operating System: Linux 5.10.17-v8+ aarch64 Raspbian
Current Operating System: Linux octopi 5.10.52-v7l+ #1441 SMP Tue Aug 3 18:11:56 BST 2021 armv7l
Kernel command line: coherent_pool=1M 8250.nr_uarts=1 snd_bcm2835.enable_compat_alsa=0 snd_bcm2835.enable_hdmi=1 video=HDMI-A-1:1024x600M@60 smsc95xx.macaddr=DC:A6:32:BA:60:50 vc_mem.mem_base=0x3ec00000 vc_mem.mem_size=0x40000000  console=tty1 root=PARTUUID=8bcff8f6-02 rootfstype=ext4 elevator=deadline fsck.repair=yes rootwait
Build Date: 19 April 2021  03:23:36PM
xorg-server 2:1.20.4-1+rpt3+deb10u3 (https://www.debian.org/support)
Current version of pixman: 0.36.0
        Before reporting problems, check http://wiki.x.org
        to make sure that you have the latest version.
Markers: (--) probed, (**) from config file, (==) default setting,
        (++) from command line, (!!) notice, (II) informational,
        (WW) warning, (EE) error, (NI) not implemented, (??) unknown.
(==) Log file: "/var/log/Xorg.0.log", Time: Wed Aug 18 17:17:14 2021
(==) Using system config directory "/usr/share/X11/xorg.conf.d"
modprobe: FATAL: Module g2d_23 not found in directory /lib/modules/5.10.52-v7l+
xinit: Unable to run program "xterm": No such file or directory
Specify a program on the command line or make sure that /usr/bin
is in your path.
xinit: connection to X server lost
waiting for X server to shut down (II) Server terminated successfully (0). Closing log file."

Solution - If I launch the Raspberry Pi Desktop (using command startx) then open terminal from the desktop and type octodash, OctoDash launches perfect and works perfect on the screen.

So is the Pi Desktop required in order to launch OctoDash? I don't want OctoDash to launch on boot, rather I want to be able to launch it manually, but can only get it to boot after desktop has been launched. I'm okay with this if that is how it needs to be, but then my additional question is, does that mean that desktop is always running in the background while OctoDash is running? I'm just concerned that if desktop, Octoprint, and OctoDash are all running at the same time off the same Pi, if it's going to cause performance errors or cause print quality issues. 

Would love to gain a little more understanding with my situation. Thank you so much for your help.

**Comment by UnchartedBull** [comment]
With a normal installation of OctoDash the raspbian desktop will never start, since OctoDash uses ratpoision by default, which is a super lightweight window manager. If you're using raspbian desktop, you'd probably need to start that beforehand. But with a 4GB Pi 4 you shouldn't run into any performance issues, even with the desktop running.

**Reply by Keele8** [reply]
Thank you for the information, that helps!
